modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -463,11 +464,9 @@ private AsyncDataCursor<InternalSqlRow> executeDdl(SqlOperationContext operation
         assert queryCancel != null;
 
         queryCancel.add(timeout -> {
-            if (!timeout) {
-                return;
-            }
+            String message = timeout ? QueryCancelledException.TIMEOUT_MSG : QueryCancelledException.CANCEL_MSG;
 
-            ret.completeExceptionally(new QueryCancelledException(QueryCancelledException.TIMEOUT_MSG));
+            ret.completeExceptionally(new QueryCancelledException(message));
```

Comment:
cancellation _must_ terminate operation, not just return error immediately. For DDL, when command is submitted to catalog there is no way to terminate the operation. Please check this condition in other places too